system/jlib/jthread.cpp

```diff
@@ -2432,10 +2432,11 @@ protected: friend class PipeWriterThread;
             if (pipeProcess != (HANDLE)-1) {
                 if (title.length())
                     PROGLOG("%s: Forcibly killing pipe process %d",title.get(),pipeProcess);
+                int signalToSend = graceful ? SIGTERM : SIGKILL;
                 if (newProcessGroup)
-                    ::kill(-pipeProcess,SIGKILL);
+                    ::kill(-pipeProcess,signalToSend);
                 else
-                    ::kill(pipeProcess,SIGKILL);            // if this doesn't kill it we are in trouble
+                    ::kill(pipeProcess,signalToSend);            // if this doesn't kill it we are in trouble
```

Comment:
if graceful, should there be a grace period, which if exceeded fires a SIGKILL anyway..?